config/config-api/src/main/java/com/thoughtworks/go/config/BasicCruiseConfig.java

```diff
@@ -335,6 +339,14 @@ private EnvironmentsConfig mergeEnvironmentConfigs() {
             return environments;
         }
 
+        private SCMs mergeScms() {
+            SCMs scms = BasicCruiseConfig.this.scms;
```

Comment:
I am not sure it is safe to modify the `scms` instance taken from xml part, we probably should clone it and then append to a new list.

Reply:
I'm having trouble understanding the logic behind some of the merge code. I'm specifically confused about the purpose of the `MergeEnvironmentConfig` and `MergePipelineConfigs`